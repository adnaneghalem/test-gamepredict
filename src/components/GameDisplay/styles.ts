import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  container: {
    flex: 1,
    paddingRight: 15,
    paddingLeft: 15,
    marginBottom: 25,
    marginTop: '7%',
  },
  item: {
    backgroundColor: '#F4F5F7',
    padding: 20,
    marginVertical: 5,
    borderRadius: 15,
    flexDirection: 'column',
  },
  headerContainer: {
    justifyContent: 'center',
    flexDirection: 'row',
    marginBottom: '5%',
  },
  teamContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  titleContainer: {
    flexDirection: 'column',
    alignSelf: 'center',
    marginBottom: 10,
    paddingRight: 10,
  },
  title: {
    fontSize: 16,
    paddingRight: 10,
    paddingBottom: 10,
  },
  id: {
    alignSelf: 'center',
  },
  score: {
    flexDirection: 'row',
  },
  titleScore: {
    fontSize: 18,
    paddingRight: 10,
  },
  tinyLogo: {
    width: 60,
    height: 60,
    alignSelf: 'center',
    resizeMode: 'contain',
  },
  button: {
    backgroundColor: '#068C85',
    alignContent: 'center',
    alignSelf: 'center',
    justifyContent: 'center',
    width: '50%',
    height: 30,
    borderRadius: 15,
  },
  buttonText: {
    textAlign: 'center',
    color: '#FFFFFF',
  },
});
