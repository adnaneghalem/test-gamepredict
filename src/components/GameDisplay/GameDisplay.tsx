import React from 'react';
import { View, Text, Image, Pressable } from 'react-native';
import dayjs from 'dayjs';
import styles from './styles';

interface GameDisplayProps {
  teamHomeName?: string;
  teamAwayName?: string;
  country?: string;
  date?: string;
  league?: string;
  id: string;
  teamHomeImage?: string;
  teamAwayImage?: string;
  onOpen: (id: string) => void | undefined;
  homeScore?: string;
  awayScore?: string;
}

function GameDisplay(props: GameDisplayProps): JSX.Element | null {
  return (
    <View key={props.id} style={styles.container}>
      <View style={styles.item}>
        <View style={styles.headerContainer}>
          <Text style={styles.title}>{props.league}</Text>
          <Text style={styles.title}>{props.country}</Text>
          <Text style={styles.title}>
            {dayjs(props.date).format('DD-MM-YY')}
          </Text>
        </View>
        <View style={styles.teamContainer}>
          <View>
            <View style={styles.titleContainer}>
              <Image
                style={styles.tinyLogo}
                source={{
                  uri: props.teamHomeImage,
                }}
              />
              <Text style={styles.title}>{props.teamHomeName}</Text>
            </View>
          </View>
          {props.homeScore && props.awayScore ? (
            <View style={styles.score}>
              <Text style={styles.titleScore}>{props.homeScore}</Text>
              <Text style={styles.titleScore}>{'-'}</Text>
              <Text style={styles.titleScore}>{props.awayScore}</Text>
            </View>
          ) : null}
          <View>
            <View style={styles.titleContainer}>
              <Image
                style={styles.tinyLogo}
                source={{
                  uri: props.teamAwayImage,
                }}
              />
              <Text style={styles.title}>{props.teamAwayName}</Text>
            </View>
          </View>
        </View>
        {!props.homeScore && !props.awayScore ? (
          <Pressable
            style={styles.button}
            onPress={() => props?.onOpen(props.id)}
          >
            <Text style={styles.buttonText}>More Info</Text>
          </Pressable>
        ) : null}
      </View>
    </View>
  );
}

export default GameDisplay;
