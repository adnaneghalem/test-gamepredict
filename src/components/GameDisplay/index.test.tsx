import React from 'react';
import renderer from 'react-test-renderer';
import { configure } from 'enzyme';
import GameDisplay from './GameDisplay';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('<GameDisplay />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<GameDisplay id="1" onOpen={() => void null} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
