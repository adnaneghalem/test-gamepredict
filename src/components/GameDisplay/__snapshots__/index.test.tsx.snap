// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GameDisplay /> renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "marginBottom": 25,
      "marginTop": "7%",
      "paddingLeft": 15,
      "paddingRight": 15,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#F4F5F7",
        "borderRadius": 15,
        "flexDirection": "column",
        "marginVertical": 5,
        "padding": 20,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
          "marginBottom": "5%",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 16,
            "paddingBottom": 10,
            "paddingRight": 10,
          }
        }
      />
      <Text
        style={
          Object {
            "fontSize": 16,
            "paddingBottom": 10,
            "paddingRight": 10,
          }
        }
      />
      <Text
        style={
          Object {
            "fontSize": 16,
            "paddingBottom": 10,
            "paddingRight": 10,
          }
        }
      >
        01-10-21
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "alignSelf": "center",
              "flexDirection": "column",
              "marginBottom": 10,
              "paddingRight": 10,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": undefined,
              }
            }
            style={
              Object {
                "alignSelf": "center",
                "height": 60,
                "resizeMode": "contain",
                "width": 60,
              }
            }
          />
          <Text
            style={
              Object {
                "fontSize": 16,
                "paddingBottom": 10,
                "paddingRight": 10,
              }
            }
          />
        </View>
      </View>
      <View>
        <View
          style={
            Object {
              "alignSelf": "center",
              "flexDirection": "column",
              "marginBottom": 10,
              "paddingRight": 10,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": undefined,
              }
            }
            style={
              Object {
                "alignSelf": "center",
                "height": 60,
                "resizeMode": "contain",
                "width": 60,
              }
            }
          />
          <Text
            style={
              Object {
                "fontSize": 16,
                "paddingBottom": 10,
                "paddingRight": 10,
              }
            }
          />
        </View>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignContent": "center",
          "alignSelf": "center",
          "backgroundColor": "#068C85",
          "borderRadius": 15,
          "height": 30,
          "justifyContent": "center",
          "width": "50%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "textAlign": "center",
          }
        }
      >
        More Info
      </Text>
    </View>
  </View>
</View>
`;
